ps_deployment/README.md

```diff
@@ -0,0 +1,121 @@
+		* cons:
+			* not copying .git to image, using default paths (/opt/app/src), maybe others which can probably require our scripts rework
+			
+	* normal image build (eg. `docker build ...`):
+		* pros:
+			* already in use in our environment
+			* no need to implement additional tool
+			* clean approach
+
+| ---------------------------------- | ----------- | ------------ |
+| image name                         | build time  | size         |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_github_base              | 4m42.531s   | 797MB        |
+| ps_normal_github_final             | 0m12.925s   | 797MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_rpm_base                 | 5m8.026s    | 804MB        |
+| ps_normal_rpm_final                | 0m15.766s   | 804MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_s2i_github_base                 | 4m38.620s   | 1.12GB       |
+| ps_s2i_github_final                | 0m10.715s   | 1.12GB       |
+| ---------------------------------- | ----------- | ------------ |
+
+### High level build flow strategies
+
+OpenShift vs. Public service
+	* OpenShift:
+		* pros:
+			* flexible
+			* no waiting in queue
+		* cons:
+			* additional maintenance
+			* additional resources = additional costs/failures because of resource limit
+	
+	* Public service:
+		* pros:
+			* our current approach - no radical changes required
+			* simple and straightforward
+			* free
+			
+		* cons:
+			* dependency on external service
+			* long queue time in case of service issue/high load
+
+#### OpenShift
+	* Source-to-Image -
+		 Source-to-Image (`sourceStrategy` in OC configuration) is using s2i tool for image deployment
+	
+	Links:
+	* [s2i details](https://docs.openshift.com/container-platform/3.11/architecture/core_concepts/builds_and_image_streams.html#source-build)
+	* [s2i strategy options](https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_strategies.html#source-to-image-strategy-options)
+	* [s2i tool blog](https://www.openshift.com/blog/create-s2i-builder-image)
+
+	* Image Stream - current configuration
+	* Custom Build - need more details
+
+## Conclusion
+
+	* Source-to-image (S2I) - no performance improvements, bigger image
+	* RPM vs github:
+		* from performance perspective basically same results 
+		* github repos should not have issues with OS version - will make deployment easier
+	* splitting image to base + final, will improve build times 
+	
+	AI:
+	
+
+## TODO
+	* trigger OC pull automatically after image is build
+	* check dockerhub alternatives
+	* using cache / forcing rebuild (e.g. pip install may install new version, yaml file is updated - force ansible to re-initiate ansible command)
+	* if base image is updated - re-trigger - how to implement this in some pipeline 
```

Comment:
For example, Docker Hub has a 'Repository Links' feature, which if set to 'Enable for Base Image' should trigger a build whenever a base image is updated. But it seems to work only for `Dockerfile`, but not for example for `Dockerfile.worker` (like [here](https://hub.docker.com/repository/docker/usercont/packit-service-worker/builds/edit))

Reply:
Maybe we could live without having anything like this ATM: images are rebuilt fairly often, so they would pick up updates in the base image relatively fast.

Reply:
thanks will look into it